crates/accelerate/src/sparse_pauli_op.rs

```diff
@@ -60,8 +62,202 @@ pub fn unordered_unique(py: Python, array: PyReadonlyArray2<u16>) -> (PyObject,
     )
 }
 
+#[derive(Clone, Copy)]
+enum Pauli {
+    I,
+    X,
+    Y,
+    Z,
+}
+
+/// A complete ZX-convention representation of a Pauli decomposition.  This is all the components
+/// necessary to construct a Qiskit-space :class:`.SparsePauliOp`, where :attr:`phases` is in the
+/// ZX convention.
+#[pyclass(module = "qiskit._accelerate.sparse_pauli_op")]
+pub struct ZXPaulis {
+    #[pyo3(get)]
+    pub z: Py<PyArray2<bool>>,
+    #[pyo3(get)]
+    pub x: Py<PyArray2<bool>>,
+    #[pyo3(get)]
+    pub phases: Py<PyArray1<u8>>,
+    #[pyo3(get)]
+    pub coeffs: Py<PyArray1<Complex64>>,
+}
+
+/// Decompose a dense complex operator into the symplectic Pauli representation in the
+/// ZX-convention.
+///
+/// This is an implementation of the "tensorized Pauli decomposition" presented in
+/// `Hantzko, Binkowski and Gupta (2023) <https://arxiv.org/abs/2310.13421>`__.
+#[pyfunction]
+pub fn decompose_dense(
+    py: Python,
+    num_qubits: usize,
+    operator: PyReadonlyArray2<Complex64>,
+    tolerance: f64,
+) -> PyResult<ZXPaulis> {
+    let size = 1 << num_qubits;
+    if operator.shape() != [size, size] {
+        return Err(PyValueError::new_err(format!(
+            "bad input shape for {} qubits: {:?}",
+            num_qubits,
+            operator.shape()
+        )));
+    }
```

Comment:
No, I was just being lazy about the `ilog2` logic in Rust space in this first draft of the code, I think, since Python-space had already computed it.  For interface simplicity, I removed the extra argument in 337c93ef03.

Reply:
Turns out that `{integer}::ilog2` is only in Rust 1.67+ - maybe that was why I was more lazy originally?  I don't remember, but anyway, dbb35e40ec puts in a Rust-1.64-compatible `ilog2`.

Reply:
Actually now our MSRV is 1.70.0 do you want to switch this to use the stdlib `ilog2`?

Reply:
Let's see - 0ebec77 was made from my phone, so let's see if I got it right.